### PR TITLE
GRBR-463: Creation of a new Validation in SignaturePolicy validation …

### PR DESCRIPTION
**Pull request by fernandoraffani**
…chain, that makes the Formal Policy Validation if it is implemented by the Custom Validation (when the Policy is in ASN1 format and signature can be validated against it).

The implementation was made so that the Formal Validation is configurable in the xml and optional, with the default value being to skip validation, so it doesn't affect all the other policy that cannot have specific elements validated against the signature.

### CODE REVIEW

File: validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java [hunk 1]
```diff
@@ -261,7 +272,12 @@ public String getPolicyDescription() {
 	public CryptographicConstraint getSignatureCryptographicConstraint(Context context) {
 		BasicSignatureConstraints basicSignature = getBasicSignatureConstraintsByContext(context);
 		if (basicSignature != null) {
-			return basicSignature.getCryptographic();
+			CryptographicConstraint sigCryptographic = basicSignature.getCryptographic();
```
Comment (davydsantos) [inline validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java]: Is this related to the SignaturePolicyChain?

File: validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java [hunk 2]
```diff
@@ -270,11 +286,20 @@ public CryptographicConstraint getSignatureCryptographicConstraint(Context conte
 	public CryptographicConstraint getCertificateCryptographicConstraint(Context context, SubContext subContext) {
 		CertificateConstraints certificateConstraints = getCertificateConstraints(context, subContext);
 		if (certificateConstraints != null) {
-			return certificateConstraints.getCryptographic();
+			CryptographicConstraint certCryptographic = certificateConstraints.getCryptographic();
```
Comment (davydsantos) [inline validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java]: Is this related to the SignaturePolicyChain?

**Review by davydsantos — CHANGES_REQUESTED** [review]
Some changes seem to be unrelated to this PR...

**Comment by fernandoraffani** [comment]
@davydsantos: I have removed the changes you have noted. 

They were due to differences between master and the version linked to validation-authority that probably need to be merged at some point. However, as they were not part of this fix, I removed them from this PR.

**Review by pauloverzini — APPROVED** [review]

**Comment by fernandoraffani** [comment]
Deprecated: created a new PR: https://github.com/davydsantos/dss/pull/10, that consists of the same changes, but in a branch created from signature-policy branch.

https://jira.corp.docusign.com/browse/GRBR-463
includes https://jira.corp.docusign.com/browse/GRBR-367

**Pull request closed without merging**